### PR TITLE
CppCheck performance fixes

### PR DESCRIPTION
**Pull request by tobbi**
This uses prefix operators instead of postfix operators where appropriate. Also: Pass by ref instead of value where appropriate.

### CODE REVIEW

File: xbmc/FileItem.cpp [hunk 1]
```diff
@@ -675,7 +675,7 @@ void CFileItem::ToSortable(SortItem &sortable, Field field) const
 void CFileItem::ToSortable(SortItem &sortable, const Fields &fields) const
 {
   Fields::const_iterator it;
-  for (it = fields.begin(); it != fields.end(); it++)
+  for (it = fields.begin(); it != fields.end(); ++it)
```
Comment (FernetMenta) [inline xbmc/FileItem.cpp]: we switched to newer style of loops: 
for (auto &field : fields)

Comment (tobbi) [inline xbmc/FileItem.cpp]: I assume I need to change all of the iterators?

Comment (FernetMenta) [inline xbmc/FileItem.cpp]: That would be really nice but we can't demand this. I would accept as is because it is already an improvement.

**Comment by MartijnKaijser** [comment]
@tobbi could you please split off the whitespace fixes?

**Comment by tobbi** [comment]
I can try, though this is something that Atom, the editor I am using, did automatically.

**Comment by MartijnKaijser** [comment]
@tobbi maybe you can update if you have some time?

**Pull request closed without merging**